storage/src/node/branch.rs

```diff
@@ -326,15 +326,30 @@ pub struct BranchNode {
     /// Element i is the child at index i, or None if there is no child at that index.
     /// Each element is (`child_hash`, `child_address`).
     /// `child_address` is None if we don't know the child's hash.
-    pub children: [Option<Child>; Self::MAX_CHILDREN],
+    pub children: [T; BranchConstants::MAX_CHILDREN],
 }
 
-impl Debug for BranchNode {
+/// Contants used in Branch. Moved out into its own struct as they are
+/// independent of the generic parameter for Branch.
+#[derive(Debug)]
+pub struct BranchConstants;
+impl BranchConstants {
+    /// The maximum number of children in a [`BranchNode`]
+    #[cfg(feature = "branch_factor_256")]
+    pub const MAX_CHILDREN: usize = 256;
+
+    /// The maximum number of children in a [`BranchNode`]
+    #[cfg(not(feature = "branch_factor_256"))]
+    pub const MAX_CHILDREN: usize = 16;
+}
+
+impl<T: ChildOption> Debug for BranchNode<T> {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         write!(f, "[BranchNode")?;
         write!(f, r#" path="{:?}""#, self.partial_path)?;
 
         for (i, c) in self.children.iter().enumerate() {
+            let c: &Option<Child> = c.as_child_option();
```

Comment:
Seems like the type should be inferred here.
```suggestion
            let c = c.as_child_option();
```